### PR TITLE
dsl extension

### PR DESCRIPTION
**Pull request by mavlyutov**

### CODE REVIEW

**Comment by mavlyutov** [comment]
summon @eroshenkoam @pupssman @baev

File: src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    public void buildFor(String buildPolicy) {
+        switch (buildPolicy.trim().toUpperCase()) {
+            case "UNSTABLE":
+                reportBuildPolicy = ReportBuildPolicy.UNSTABLE;
+                break;
+            case "UNSUCCESSFUL":
+                reportBuildPolicy = ReportBuildPolicy.UNSUCCESSFUL;
+                break;
+            default:
+                reportBuildPolicy = ReportBuildPolicy.ALWAYS;
```
Comment (pupssman) [inline src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java]: I'd seek ways to enforce correctness in this case. A typo in `UNSUCCESSFUL` is too common not to treat it.

File: pom.xml [hunk 2]
```diff
@@ -114,6 +114,18 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>job-dsl</artifactId>
+            <version>1.35</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>matrix-project</artifactId>
```
Comment (pupssman) [inline pom.xml]: Can't see why we'd need that kind of dependency. `matrix` is only mentioned in this line.

**Comment by pupssman** [comment]
Looks nice otherwise)

File: pom.xml (hunk 2 above)
Comment (mavlyutov) [inline pom.xml]: dsl-plugin require jenkins version >= 1.565
but in that version matrix-project has been cut out from jenkins-core, so we need to depend on matrix-project

File: src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    public ReportVersionPolicy getReportVersionPolicy() {
+        return reportVersionPolicy;
+    }
+
+    public ReportBuildPolicy getReportBuildPolicy() {
+        return reportBuildPolicy;
+    }
+
+    public boolean getIncludeProperties() {
+        return includeProperties == null || includeProperties;
```
Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java]: maybe move this check to setter?

File: src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java (hunk 1 above)
Comment (mavlyutov) [inline src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java]: ideas?

Comment (mavlyutov) [inline src/main/java/ru/yandex/qatools/allure/jenkins/AllureDslExtension.java]: I'll renamed it to 'FAILURE' like those guys do:
https://github.com/jenkinsci/job-dsl-plugin/wiki/Job-reference#upstream

**Comment by baev** [comment]
LGTM

**Pull request merged**